src/temporal/t.rast.boxplot/t.rast.boxplot.py

```diff
@@ -0,0 +1,620 @@
+# %option
+# % key: fontsize
+# % type: integer
+# % label: Font size
+# % description: Default font size
+# % guisection: Plot format
+# % required: no
+# %end
+
+# %option
+# % key: date_format
+# % type: string
+# % label: Date format
+# % description: Set date format (see https://strftime.org/ for options)
+# % guisection: Plot format
+# % required: no
+# %end
+
+# %option
+# % key: boxplot_width
+# % type: double
+# % label: Boxplot width
+# % description: The width of the boxplots
+# % required: no
+# % guisection: Boxplot format
+# %end
+
+# %options G_OPT_CN
+# % key: bxcolor
+# % type: string
+# % label: Boxplot color
+# % description: Color of the boxplots. See manual page for color notation options.
+# % required: no
+# % guisection: Boxplot format
+# %end
+
+# %option
+# % key: flier_marker
+# % type: string
+# % label: Flier marker
+# % description: Set flier marker (see https://matplotlib.org/stable/api/markers_api.html for options)
+# % required: no
+# % guisection: Boxplot format
+# %end
+
+# %option
+# % key: flier_size
+# % type: string
+# % label: Flier size
+# % description: Set the flier size
+# % required: no
+# % guisection: Boxplot format
+# %end
+
+# %option G_OPT_C
+# % key: flier_color
+# % label: Flier color
+# % description: Set the flier color
+# % required: no
+# % guisection: Boxplot format
+# %end
+
+
+import sys
+import atexit
+import uuid
+from subprocess import PIPE
+from dateutil import parser
+import matplotlib.dates as mdates
+import grass.script as gs
+from grass.pygrass.modules import Module
+
+
+clean_layers = []
+
+
+def create_unique_name(name):
+    """Generate a temporary name which contains prefix
+    Store the name in the global list.
+    """
+    return name + str(uuid.uuid4().hex)
+
+
+def create_temporary_name(prefix):
+    """Create temporary file name"""
+    tmpf = create_unique_name(prefix)
+    clean_layers.append(tmpf)
+    return tmpf
+
+
+def cleanup():
+    """Remove temporary maps specified in the global list"""
+    cleanrast = list(reversed(clean_layers))
+    for layername in cleanrast:
+        ffile = gs.find_file(
+            name=layername, element="cell", mapset=gs.gisenv()["MAPSET"]
+        )
+        if ffile["file"]:
+            Module(
+                "g.remove",
+                flags="f",
+                type="raster",
+                name=layername,
+                quiet=True,
+            )
+        vfile = gs.find_file(
+            element="vector", name=layername, mapset=gs.gisenv()["MAPSET"]
+        )
+        if vfile["file"]:
+            Module(
+                "g.remove",
+                flags="f",
+                type="vector",
+                name=layername,
+                quiet=True,
+            )
+
+
+def strip_mapset(name):
+    """Strip Mapset name and '@' from map name
+    >>> strip_mapset('elevation@PERMANENT')
+    elevation
+    """
+    if "@" in name:
+        return name.split("@")[0]
+    return name
+
+
+def main(options, flags):
```

Comment:
Could you split this long function into many small functions, please?

Reply:
I'll have a look at this. I am not sure what would make sense to split off as a separate function.

Reply:
Yes It makes sense in terms of readability and simplicity of code. If you need any suggestions, I will help you with that.

Reply:
Thanks, I will make a first attempt, but will probably hold you to your offer :-).

Reply:
I splitted the code in various functions.